### PR TITLE
Add basic code-stile settings

### PR DESCRIPTION
**Pull request by malyshevhen**
## Add basic code-stile maven plugins:

- `maven-surefire-plugin`
- `maven-checkstyle-plugin` 
- `jacoco-maven-plugin`
  - classes coverage limit: 100%
  - methods coverage limit: 80%
  - lines coverage limit: 60%
- config-files for `maven-checkstyle-plugin` in path: ./config/checkstyle/*.xml
- change the `.editorconfig` file following the `sun_checkstyle.xml`

### CODE REVIEW

**Comment by Anatolii-Omelchenko** [comment]
Fix PR name.

**Comment by malyshevhen** [comment]
Done

File: config/checkstyle/checkstyle-suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<!DOCTYPE suppressions PUBLIC
+    "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+    "https://www.puppycrawl.com/dtd/suppressions_1_1.dtd">
+
+<suppressions>
+    <!--  suppress package-info.java check  -->
+    <suppress files="src[/\\]main[/\\]java[/\\].*" checks="JavadocPackage"/>
+    <suppress files="src[/\\]test[/\\]java[/\\].*" checks="JavadocPackage"/>
+
```
Comment (Scream274) [inline config/checkstyle/checkstyle-suppressions.xml:10]:
```suggestion
<suppressions>
    <!--  suppress package-info.java check  -->
    <suppress files="src[/\\]main[/\\]java[/\\].*" checks="JavadocPackage"/>
    <suppress files="src[/\\]test[/\\]java[/\\].*" checks="JavadocPackage"/>
    <suppress files="src[/\\]test[/\\]java[/\\].*" checks="JavadocPackage"/>

```

Comment (Scream274) [inline config/checkstyle/checkstyle-suppressions.xml:10]:
```suggestion
System.out.println("Hello world!");

```

**Review by Anatolii-Omelchenko — APPROVED** [review]

**Pull request merged**